### PR TITLE
New examples: hosts and virtualmachines. WIP

### PR DESCRIPTION
**Pull request by amandahla**
TODO:
- [x] Adding new examples for collect hosts and virtual machines information.

### CODE REVIEW

**Comment by dougm** [comment]
Thanks @amandahla 

Looks like these examples would be a good place to use the newer ContainerView APIs.  This reduces the number of round trips quite a bit.  An untested example below.  What do you think?

```go
	m := view.NewManager(c.Client)

	v, err := m.CreateContainerView(ctx, c.ServiceContent.RootFolder, []string{"VirtualMachine"}, true)
	if err != nil {
		exit(err)
	}

	var vms []mo.VirtualMachine

	err = v.RetrieveWithFilter(ctx, kind, []string{"summary"}, &vms, nil)
	if err != nil {
		exit(err)
	}

	_ = v.Destroy(ctx)
```

**Comment by amandahla** [comment]
@dougm Thanks, I'll change them! :-)

**Comment by amandahla** [comment]
@dougm I updated datastores also, ok? Did I get the ideia correctly? When we use view package, they get from all datacenters, right?

File: examples/hosts/main.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
+/*
+Copyright (c) 2015 VMware, Inc. All Rights Reserved.
```
Comment (dougm) [inline examples/hosts/main.go]: Let's me these 2017

File: examples/virtualmachines/main.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	// Get summary property (VirtualMachineSummary)
+	// Reference for this data object:
+	// https://www.vmware.com/support/developer/vc-sdk/visdk25pubs/ReferenceGuide/vim.vm.Summary.html
+
+	var vms []mo.VirtualMachine
+	err = v.Retrieve(ctx, []string{"VirtualMachine"}, []string{"summary"}, &vms)
+	if err != nil {
+		exit(err)
+	}
+
```
Comment (dougm) [inline examples/virtualmachines/main.go]: At this point, you have all the data in memory, so there's no need for the go routines.  Just the for loop + print is fine.

Comment (amandahla) [inline examples/virtualmachines/main.go]: Hi @dougm  Changed now :)

**Comment by dougm** [comment]
@amandahla great, thanks!

**Pull request merged**